lib/linter-ruby.coffee

```diff
@@ -1,33 +1,54 @@
-linterPath = atom.packages.getLoadedPackage("linter").path
-Linter = require "#{linterPath}/lib/linter"
-
-class LinterRuby extends Linter
-  # The syntax that the linter handles. May be a string or
-  # list/tuple of strings. Names should be all lowercase.
-  @syntax: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec']
-
-  # A string, list, tuple or callable that returns a string, list or tuple,
-  # containing the command line (with arguments) used to lint.
-  cmd: 'ruby -wc'
-
-  executablePath: null
-
-  errorStream: 'stderr'
-
-  linterName: 'ruby'
-
-  # A regex pattern used to extract information from the executable's output.
-  regex:
-    '.+:(?<line>\\d+):((?<warning> warning:)|(?<error>))(?<message>.+)'
-
-  constructor: (editor)->
-    super(editor)
-
-    @rubyLinter = atom.config.observe 'linter-ruby.rubyExecutablePath', =>
-      @executablePath = atom.config.get 'linter-ruby.rubyExecutablePath'
-
-  destroy: ->
-    super
-    @rubyLinter.dispose()
-
-module.exports = LinterRuby
+child_process = require 'child_process'
+path = require 'path'
+
+module.exports = LinterRuby =
+  config:
+    executablePath:
+      default: 'ruby'
+      type: 'string'
+
+  activate: ->
+    unless atom.packages.getLoadedPackages 'linter-plus'
+      atom.notifications.addError(
+        '[linter-ruby] `linter-plus` package not found, please install it')
+
+  provideLinter: -> {
+    grammarScopes: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec']
+    scope: 'file'
+    lint: @lint
+    lintOnFly: false
+  }
+
+  lint: (TextEditor) ->
+    regex = ///
+      (.+): #File with issue
+      (\d+):\s+ #Line with issue
+      (\w+(?:\s(?:error|warning))?) #Issue type
+      (?:,\s) #A command and space seperating the type and message
+      (.+) #Message explaining the issue
+    ///
+    return new Promise (Resolve) ->
+      filePath = TextEditor.getPath()
+      if filePath # Files that have not be saved
```

Comment:
If someone else hasn't changed, I wrote the line myself in `editor-linter.coffee` to ignore the files that have no filePath and wait for user to save them somewhere and lint them next time.